### PR TITLE
Minor bugfixes

### PR DESCRIPTION
**Pull request by OpatrilPeter**
Fixes several cases where behaviour was determined to be obviously incorrect.

### CODE REVIEW

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp [hunk 1]
```diff
@@ -47,9 +47,9 @@ namespace SuperiorMySqlpp
         constexpr inline void initializeNullable(MYSQL_BIND& binding, Nullable<T>& nullable)
         {
             // Checks if pointer of C client pointer to their custom bool bool variant is compatible with bool*
-            // TODO: It would be wise to check also sizeof(my_bool) == sizeof(bool)
-            static_assert(sizeof(decltype(binding.is_null)) == sizeof(decltype(&nullable.detail_getNullRef())),
-                "Pointers to null indicators must have same size.");
+            static_assert(sizeof(decltype(binding.is_null)) == sizeof(decltype(&nullable.detail_getNullRef()))
+                && sizeof(decltype(*binding.is_null)) == sizeof(decltype(nullable.detail_getNullRef())),
```
Comment (majcl) [inline include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp]: Is it really needed? Nullable flag is initializable even without second condition, isnt it?

File: include/superior_mysqlpp/types/nullable.hpp [hunk 2]
```diff
@@ -371,7 +371,7 @@ class Nullable : public detail::NullableBase
         if (engaged && other.engaged)
         {
             ::std::swap(payload, other.payload);
-            ::std::swap(null, null);
+            ::std::swap(null, other.null);
```
Comment (majcl) [inline include/superior_mysqlpp/types/nullable.hpp:374]: ups :-O

**Review by majcl — CHANGES_REQUESTED** [review]

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp (hunk 1 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp]: `binding.is_null` has type `my_bool*`, &nullable.detail_getNullRef() has type `bool*`, we feed our pointer to C client.
So, we clearly need sanity checks that `my_bool` has same size as `bool`, the second condition. That said, we perhaps do NOT need the first one, as the second one likely implies the first one, that their respective pointers have matching sizes. Opinions?

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp [hunk 3]
```diff
@@ -47,9 +47,9 @@ namespace SuperiorMySqlpp
         constexpr inline void initializeNullable(MYSQL_BIND& binding, Nullable<T>& nullable)
         {
             // Checks if pointer of C client pointer to their custom bool bool variant is compatible with bool*
```
Comment (smradCZ) [inline include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp]: Doubled `bool bool`

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp (hunk 1 above)
Comment (smradCZ) [reply include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp]: I'd prefer remove the first part of condition, checking pointer sizes. The second condition make sense to me.

**Review by smradCZ — CHANGES_REQUESTED** [review]

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp (hunk 3 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp]: Oh yes, the `bool bool` type, you must have heard about it. Just like `long long` means "it's very long", it's meant to be used only when your bool is really booleany, no mere disguised integer ... that, or I just made a typo. 😛

**Review by smradCZ — DISMISSED** [review]

File: include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp (hunk 1 above)
Comment (OpatrilPeter) [reply include/superior_mysqlpp/prepared_statements/initialize_bindings.hpp]: After further offline discussion, we ended up keeping both conditions, main argument being that the original condition was done by original lib author, although without commentary what real situation it was meant to cover.

**Review by majcl — APPROVED** [review]

**Pull request merged**